### PR TITLE
Change membership date format

### PR DESCRIPTION
**Pull request by smnkrt**
Change Membership `starts_at` and `ends_at` field types to datetime.

### CODE REVIEW

**Comment by mateuszgorniak** [comment]
There's a problem with validation now (or at least with values in factories) - please look at specs.

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -170,7 +170,7 @@ def time(year, month, day)
   describe "#booked_memberships" do
     let!(:project_current) { create(:project, name: "google", potential: true) }
     let!(:project_next) { create(:project, name: "facebook", potential: false) }
-    let!(:project_future) { create(:project, name: "estimote", potential: false) }
+    let!(:project_future) { create(:project, name: "estimote", potential: false, end_at: nil) }
```
Comment (netguru-hound) [inline spec/models/user_spec.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: spec/models/membership_spec.rb [hunk 2]
```diff
@@ -188,7 +187,7 @@
         new_starts_at = membership.starts_at + 1.day
         expected_notification = "Time span for *#{membership.user.last_name} #{membership.user.first_name}*"
         expected_notification += " in *#{membership.project.name}* has been changed."
-        expected_notification += "\nStarts at changed from _#{membership.starts_at}_ to _#{new_starts_at}_."
+        expected_notification += "\nStarts at changed from _#{membership.starts_at.to_date}_ to _#{new_starts_at.to_date}_."
```
Comment (netguru-hound) [inline spec/models/membership_spec.rb]: Line is too long. [124/100]

File: spec/models/membership_spec.rb [hunk 3]
```diff
@@ -160,25 +159,25 @@
...
       it 'notifies with proper message if starts_at and ends_at changed' do
         new_starts_at = membership.starts_at + 1.day
-        new_ends_at = Date.current + 2.days
+        new_ends_at = DateTime.current + 2.days
         expected_notification = "Time span for *#{membership.user.last_name} #{membership.user.first_name}*"
         expected_notification += " in *#{membership.project.name}* has been changed."
-        expected_notification += "\nStarts at changed from _#{membership.starts_at}_ to _#{new_starts_at}_."
-        expected_notification += "\nEnds at changed from _not specified_ to _#{new_ends_at}_."
+        expected_notification += "\nStarts at changed from _#{membership.starts_at.to_date}_ to _#{new_starts_at.to_date}_."
+        expected_notification += "\nEnds at changed from _not specified_ to _#{new_ends_at.to_date}_."
```
Comment (netguru-hound) [inline spec/models/membership_spec.rb]: Line is too long. [102/100]

File: spec/models/membership_spec.rb [hunk 4]
```diff
@@ -160,25 +159,25 @@
...
 
       it 'notifies with proper message if starts_at and ends_at changed' do
         new_starts_at = membership.starts_at + 1.day
-        new_ends_at = Date.current + 2.days
+        new_ends_at = DateTime.current + 2.days
         expected_notification = "Time span for *#{membership.user.last_name} #{membership.user.first_name}*"
         expected_notification += " in *#{membership.project.name}* has been changed."
-        expected_notification += "\nStarts at changed from _#{membership.starts_at}_ to _#{new_starts_at}_."
-        expected_notification += "\nEnds at changed from _not specified_ to _#{new_ends_at}_."
+        expected_notification += "\nStarts at changed from _#{membership.starts_at.to_date}_ to _#{new_starts_at.to_date}_."
```
Comment (netguru-hound) [inline spec/models/membership_spec.rb]: Line is too long. [124/100]

File: spec/models/membership_spec.rb [hunk 5]
```diff
@@ -160,25 +159,25 @@
       it 'notifies with proper message of starts_at and ends_at set' do
         expected_notification = "*#{membership_with_ends_at.user.last_name} #{membership_with_ends_at.user.first_name}*"
         expected_notification += " has been added to *#{membership_with_ends_at.project.name}* since"
-        expected_notification += " _#{membership_with_ends_at.starts_at}_ to _#{membership_with_ends_at.ends_at}_."
+        expected_notification += " _#{membership_with_ends_at.starts_at.to_date}_ to _#{membership_with_ends_at.ends_at.to_date}_."
```
Comment (netguru-hound) [inline spec/models/membership_spec.rb]: Line is too long. [131/100]

File: spec/models/membership_spec.rb [hunk 6]
```diff
@@ -144,14 +143,14 @@
...
-    describe "#notify_slack_on_create" do
+  context 'notifies on slack if created' do
+    describe '#notify_slack_on_create' do
       let(:membership) { build(:membership, starts_at: 5.months.ago, ends_at: nil) }
       let(:membership_with_ends_at) { build(:membership, starts_at: 5.months.ago, ends_at: 1.day.ago) }
 
       it 'notifies with proper message if only starts_at set' do
         expected_notification = "*#{membership.user.last_name} #{membership.user.first_name}*"
-        expected_notification += " has been added to *#{membership.project.name}* since _#{membership.starts_at}_."
+        expected_notification += " has been added to *#{membership.project.name}* since _#{membership.starts_at.to_date}_."
```
Comment (netguru-hound) [inline spec/models/membership_spec.rb]: Line is too long. [123/100]

File: spec/models/concerns/notification/membership/updated_spec.rb [hunk 7]
```diff
@@ -15,12 +15,12 @@
...
     it 'returns proper string when starts_at and ends_at changed' do
       new_starts_at = membership_without_ends_at.starts_at + 1.day
-      new_ends_at = Date.current + 2.days
+      new_ends_at = DateTime.current + 2.days
 
       expected_notification = "Time span for *#{user.last_name} #{user.first_name}*"\
         " in *#{project.name}* has been changed."\
-        "\nStarts at changed from _#{membership_without_ends_at.starts_at}_ to _#{new_starts_at}_."\
-        "\nEnds at changed from _not specified_ to _#{new_ends_at}_."
+        "\nStarts at changed from _#{membership_without_ends_at.starts_at.to_date}_ to _#{new_starts_at.to_date}_."\
```
Comment (netguru-hound) [inline spec/models/concerns/notification/membership/updated_spec.rb]: Line is too long. [116/100]

File: spec/models/concerns/notification/membership/created_spec.rb [hunk 8]
```diff
@@ -11,14 +11,14 @@
   describe '#message' do
     it 'returns proper string when ends_at value is present' do
       expected_notification = "*#{user.last_name} #{user.first_name}* has been added to"\
-      " *#{project.name}* since _#{membership.starts_at}_ to _#{membership.ends_at}_."
+      " *#{project.name}* since _#{membership.starts_at.to_date}_ to _#{membership.ends_at.to_date}_."
```
Comment (netguru-hound) [inline spec/models/concerns/notification/membership/created_spec.rb]: Line is too long. [102/100]

File: app/models/user.rb [hunk 9]
```diff
@@ -61,7 +61,10 @@ class User < ActiveRecord::Base
   scope :available, -> { where.not(id: unavailable.select(:id)) }
   scope :unavailable, -> do
     joins(memberships: :project).where("lower(projects.name) = 'unavailable'")
-      .where('(memberships.ends_at IS NULL) OR (memberships.ends_at >= ?)', Time.current)
+      .where(
+        '(memberships.ends_at IS NULL) OR (memberships.ends_at >= ?)',
+        DateTime.current.beginning_of_day
```
Comment (mateuszgorniak) [inline app/models/user.rb]: or `Date.today.to_datetime` :-)

**Comment by mateuszgorniak** [comment]
:+1:

**Pull request merged**